Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1

```diff
@@ -359,24 +359,105 @@ function Publish-UpgradedServiceFabricApplication
             return
         }
 
-        $upgradeStatus = Wait-ServiceFabricApplicationUpgradeAction -ApplicationName $ApplicationName
+        $unhealthyEvaluationMessageFunction = {
+            param(
+                [object]$UnhealthyEvaluations,
 
-        if ($UnregisterUnusedVersions)
-        {
-            Write-Host (Get-VstsLocString -Key SFSDK_UnregisterUnusedVersions)
-            foreach ($registeredAppType in Get-ServiceFabricApplicationTypeAction -ApplicationTypeName $names.ApplicationTypeName)
+                [string]$Indentation
```

Comment:
Could this have worked without passing $indentation if you prepend `n or `t on the return value of children unhealthyEvaluationMessageFunction ?

Reply:
I think the existing one is a better approach as we dealing with an array of children unhealthy evaluations, thus if we prepend indentation in parent then we might need to iterate through multiple lines.